pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/ZKMetadataStore.java

```diff
@@ -201,10 +202,13 @@ protected void batchOperation(List<MetadataOp> ops) {
                                         Collectors.groupingBy(MetadataOp::getType, Collectors.summingInt(op -> 1)))
                                 .entrySet().stream().map(e -> e.getValue() + " " + e.getKey().name() + " entries")
                                 .collect(Collectors.joining(", "));
+                        List<Pair> opsForLog = ops.stream().map(op -> Pair.of(op.getPath(), op.size()))
+                                .collect(Collectors.toList());
```

Comment:
Since this list is only for logging, I think it would be more useful to build a String directly in the desired format. It might be good to limit how many entries are logged and split to multiple log messages in case there could be a large amount of ops. Is the size of the String going to be a problem?

Reply:
`java.util.StringJoiner` (or `Collectors.joining`) could be used instead.

Reply:
Added a filter `op -> op.size > 256K`